application/main.py

```diff
@@ -682,7 +682,12 @@ def download_model(decoded, id):
 @application.route('/delete/<id>', methods=['GET'])
```

Comment:
This should not be a GET request, but a POST request. See https://stackoverflow.com/questions/46585/when-do-you-use-post-and-when-do-you-use-get

Even then with POST, this is still at risk of CSRF https://testdriven.io/blog/csrf-flask/ (which should be solved, but let's create a new issue for that on the roadmap)